script.py

```diff
@@ -0,0 +1,269 @@
+        .toPandas()
+        .assign(pdbStructureId=lambda df: df.pdbCompoundId.apply(get_structure))
+    )
+
+    logging.info('FINISHED: Get pdbStructureId on pdbe API.')
+
+    inchikey_molecule_structure_joined_pd_df.to_csv('output_files/structure_of_molecules.csv', index=False, header=True)
+
+    # Convert Spark DataFrame into Pandas DataFrame
+    inchikey_molecule_structure_joined_spark_df = spark.createDataFrame(inchikey_molecule_structure_joined_pd_df)
+
+    # ----- STATISTICS -----
+    total_nb_struct_df = (
+        inchikey_molecule_structure_joined_spark_df
+        .withColumn('pdbStructureId', f.explode('pdbStructureId'))
+        .select('pdbStructureId')
+        .distinct()
+    )
+
+    nb_molecule_without_struct_df = (
+        inchikey_molecule_structure_joined_spark_df
+        .filter(f.size('pdbStructureId') == 0)
+        .distinct()
+    )
+
+    logging.info(f'RESULT: Number of structures (pdbId): {total_nb_struct_df.count()}')
+    logging.info(f'RESULT: Number of molecule without structure: {nb_molecule_without_struct_df.count()}')
+
+    # Count number of structure per molecule
+    inchikey_molecule_structure_joined_spark_df = (
+        inchikey_molecule_structure_joined_spark_df
+        .withColumn(
+            "nbPdbStructure",
+            f.size(f.col("pdbStructureId"))
+        )
+        .orderBy(f.desc("nbPdbStructure"))
+    )
+
+    # Save into a CSV
+    inchikey_molecule_structure_joined_spark_df.toPandas().to_csv("output_files/molecule_structure_target.csv", index=False, header=True)
+
+    # ----- GET TARGET ID -----
+    pdb_target_df = (
+            spark.read.csv(args.ensembl, sep=',', header=True, comment='#')
+            .select(
+                f.col('PDB').alias('pdbStructureId'),
+                f.col('GENE_ID').alias('geneId')
+            )
+            .groupby("pdbStructureId")
+            .agg(
+                f.collect_set(f.col("geneId")).alias("geneIds")
+            )
+            .distinct()
+            .persist()
+    )
+
+    nb_target_pd_df = (
+        pdb_target_df
+        .withColumn('geneIdsExploded', f.explode('geneIds'))
+    )
+
+    nb_human_target_df = (
+        pdb_target_df
+        .withColumn('geneIdsExploded', f.explode('geneIds'))
+        .filter(f.col('geneIdsExploded').startswith('ENSG'))
+    )
+
+    logging.info(f'RESULT: Number of targets (ensemblId): {nb_target_pd_df.count()}')
+    logging.info(f'RESULT: Number of human targets: {nb_human_target_df.count()}')
+    logging.info(f'RESULT: Number of non-human targets: {nb_target_pd_df.count() - nb_human_target_df.count()}')
+
+    inchikey_molecule_structure_target_joined_spark_df = (
+        inchikey_molecule_structure_joined_spark_df
+        .select("chemblId", "pdbCompoundId", f.explode("pdbStructureId").alias("pdbStructureId"))
+        .join(pdb_target_df, on="pdbStructureId", how="inner") # Only keep the structures mapped by ensembl (inner)
+    )
+
+    df = (
```

Comment:
Please always try to avoid such variable names. I see certain cases it is not easy to find the proper variable name to reflect the content/function, but if all we know about the structure that it is a dataframe... well, that's hard to follow.